lib/infer/return.js

```diff
@@ -29,9 +31,14 @@ module.exports = function () {
     if (t.isFunction(fn) &&
       fn.returnType &&
       fn.returnType.typeAnnotation) {
-      comment.returns = [{
-        type: flowDoctrine(fn.returnType.typeAnnotation)
-      }];
+      var returnType = flowDoctrine(fn.returnType.typeAnnotation);
+      if (comment.returns) {
```

Comment:
We won't get to this point if it isn't, since we return early on L20 if `comment.returns.length` is 0.